### ISSUE TITLE
Big Sur 11.1 Macbook Pro M1 not HiDPI

### ISSUE THREAD

**Issue by fernangcia** [xzhih/one-key-hidpi#164]
Hi xzhih, thank you for your work.

Trying to enable HiDPI on my external monitor (Dell U2515h) at 2560x1440 with no success.
The script seems to make changes but SwitchResx does not enable HiDPI for any resolution.
Could it be because of Apple Silicon processor?
Thanks.
![CleanShot 2021-01-04 at 10 22 22](https://user-images.githubusercontent.com/12585812/103520037-cef1e980-4e76-11eb-8a75-c909854f08e5.png)

**Comment by xzhih** [comment]
#159 
#157 

Yes, M1 is not supported now.
Need Apple update.

**Comment by zzm88** [comment]
Is there any workaround before the update?

**Comment by akswalia23** [comment]
yeah any work around at all?

**Comment by WilliamHoo** [comment]
i updated to big sur 11.2 this morning, still cannot scale the display on my external dell 32'' monitor (2k). Does anyone knows how to fix this problem?

@xzhih please let me know if you wish to know anything from me to fix this problem. It really hurts my eyes now. :(

**Comment by akswalia23** [comment]
yeah me too

**Comment by Swiftpaws** [comment]
On M1 macs creating custom resolutions is not possible due to EDID read issues. https://www.reddit.com/r/mac/comments/k7pkrp/your_help_is_needed/

There is no workaround as of now, Apple has to provide a fix

**Comment by cooper-lyt** [comment]
SwitchResX 4.11.1a3  can creating custom resolutions .

**Comment by cooper-lyt** [comment]
SwitchResX 4.11.1a3 can  creating custom resolutions.
<img width="1380" alt="iShot2021-02-18 10 13 17" src="https://user-images.githubusercontent.com/4219608/108295212-4636cf00-71d2-11eb-9915-2526b98fae05.png">
<img width="1321" alt="iShot2021-02-18 10 17 12" src="https://user-images.githubusercontent.com/4219608/108295360-7bdbb800-71d2-11eb-8d6b-be1a345cb576.png">

**Comment by Swiftpaws** [comment]
@cooper-lyt does your monitor have 1440p native res? Im trying to create a hidpi 1080p resultion but the status always is "not active - invalid?"

**Comment by Swiftpaws** [comment]
Yes but it is still limited because of Apple limitations. For example creating HiDpi 1080p resolutions (on a 1440p panel) is not possible.

**Comment by ilovezans** [comment]
I have a M1 MacBook Air, and a Dell P2419HC 1920x1080p Monitor. The text is blurry and weird, games etc are normal. Will this work in the future to fix that?

**Comment by remoteweb** [comment]
> 4.11.1a3

There is only 4.11.1 version available for M1 and custom resolutions don't seem available. How did you manage to enable these tabs (display information, custom resolutions)

**Comment by remoteweb** [comment]
> SwitchResX 4.11.1a3 can creating custom resolutions.
> <img alt="iShot2021-02-18 10 32 23" width="1301" src="https://user-images.githubusercontent.com/4219608/108296512-a4fd4800-71d4-11eb-9616-3e702cb89d4a.png">

There is only 4.11.1 version available for M1 and custom resolutions don't seem available. How did you manage to enable these tabs (display information, custom resolutions)

**Comment by khoatdd** [comment]
> > SwitchResX 4.11.1a3 can creating custom resolutions.
> > <img alt="iShot2021-02-18 10 32 23" width="1301" src="https://user-images.githubusercontent.com/4219608/108296512-a4fd4800-71d4-11eb-9616-3e702cb89d4a.png">
> 
> There is only 4.11.1 version available for M1 and custom resolutions don't seem available. How did you manage to enable these tabs (display information, custom resolutions)

You need to download beta version, not stable one

**Comment by khoatdd** [comment]
> > SwitchResX 4.11.1a3 can creating custom resolutions.
> > <img alt="iShot2021-02-18 10 32 23" width="1301" src="https://user-images.githubusercontent.com/4219608/108296512-a4fd4800-71d4-11eb-9616-3e702cb89d4a.png">
> 
> There is only 4.11.1 version available for M1 and custom resolutions don't seem available. How did you manage to enable these tabs (display information, custom resolutions)

You need to use beta version, not stable one

**Comment by lei2rock** [comment]
> > > SwitchResX 4.11.1a3 can creating custom resolutions.
> > > <img alt="iShot2021-02-18 10 32 23" width="1301" src="https://user-images.githubusercontent.com/4219608/108296512-a4fd4800-71d4-11eb-9616-3e702cb89d4a.png">
> > 
> > 
> > There is only 4.11.1 version available for M1 and custom resolutions don't seem available. How did you manage to enable these tabs (display information, custom resolutions)
> 
> You need to use beta version, not stable one

I can create the custom resolution but cannot make this active with beta version.

**Comment by remoteweb** [comment]
@lei2rock @khoatdd I talked with Stephane, creator of SwitchResX and there is no way we can enable 1080p HiDPi for 1440p external displays as we used to, with older intel based Mac products. I would advice we all submit a suggestion for this on https://feedbackassistant.apple.com . Creator told me that this feature was initially enabled in M1 BigSur beta version which Apple blocked afterwards.

At the moment this HiDPi feature works only on 4k external monitors.

**Comment by tvhung83** [comment]
@remoteweb thank you for the info. That's sad to know. I've submitted many feedbacks to Apple, and not a single update from them, for months :(

**Comment by ConfusedInterval** [comment]
https://forums.macrumors.com/threads/scale-1680x1050-on-rmb-without-an-app.1873910/post-29666683 This guy successfully created and activated custom resolution using newest 11.3 beta. Maybe 1080p HiDPI would(/will) work on Big Sur 11.3.

**Comment by nildeveloper** [comment]
期待解决  眼都要废了

**Comment by remoteweb** [comment]
Yes creator of SwitchResX told me that this is doable. And i did it also for my macbook. However the HiDPi is not working ... and the display is kind of blurry. @pjuhas

**Comment by leocho1989** [comment]
So download new version SwitchresX, add a custom resolution like 5120x2144 into internal display, remember to save it and restart the mac. Then you can mirror the exterior display to internal display in display settings. Then it should be hidpi on external display too. I am using m1 macboom air on 11.3 and 11.4 beta. This should be the solution so far. Hope this can help. Been using like this for weeks.

**Comment by tvhung83** [comment]
> So download new version SwitchresX, add a custom resolution like 5120x2144 into internal display, remember to save it and restart the mac. Use SwitchresX set the internal resolution to the new one. Then you can mirror the exterior display to internal display in display settings. Then it should be hidpi on external display too. I am using m1 macboom air on 11.3 and 11.4 beta on a 34 ultrawide. This should be the solution so far. Hope this can help. Been using like this for weeks.

What is your external monitor, ie. brand and resolution? Did you have the issue of "Not activate - Invalid"?

> Then you can mirror the exterior display to internal display in display settings.

Can you give more details, screenshots would help. The reason I ask is, I'm using mac mini, there is no internal display. Thanks in advance!

**Comment by leocho1989** [comment]
> > So download new version SwitchresX, add a custom resolution like 5120x2144 into internal display, remember to save it and restart the mac. Use SwitchresX set the internal resolution to the new one. Then you can mirror the exterior display to internal display in display settings. Then it should be hidpi on external display too. I am using m1 macboom air on 11.3 and 11.4 beta on a 34 ultrawide. This should be the solution so far. Hope this can help. Been using like this for weeks.
> 
> What is your external monitor, ie. brand and resolution? Did you have the issue of "Not activate - Invalid"?
> 
> > Then you can mirror the exterior display to internal display in display settings.
> 
> Can you give more details, screenshots would help. The reason I ask is, I'm using mac mini, there is no internal display. Thanks in advance!

Mine is LG 34 ultrawide  3440x 1440
for mac mini it doesn’t have an internal display, so probably it can use a 4k edid emulator to work as an internal display.

 
![93A55AF8-DAE5-4F3D-A059-67BBA25083B7](https://user-images.githubusercontent.com/78505232/116021887-f9220f80-a5fd-11eb-96da-e701d60329de.png)

**Comment by usmajeed** [comment]
Yep, agree with above. Someone posted something similar on Macrumors. More detailed.

https://forums.macrumors.com/threads/custom-hidpi-resolution-for-qhd-1440p-uw-working-on-m1.2295146/post-29863166

**Comment by JM-Mendez** [comment]
For those using m1 macs, this just worked for me on a macbook


> SwitchResX -> create a custom resolution for INTERNAL display. Make it 2x the resolution you want, i.e. 3840x2160 for 1080p. Save it and reboot.
>
> Then, connect your laptop up to your monitor. Select the SwitchResX icon in the menu bar at the top. Activate the custom HIDPI res for your internal monitor.

Once you've done that, press "turn video mirroring on", also in the switchres menu bar.
https://forums.macrumors.com/threads/scale-1680x1050-on-rmb-without-an-app.1873910/page-17?post=29863160#post-29863160

**Comment by sunjianan9900** [comment]
> > > So download new version SwitchresX, add a custom resolution like 5120x2144 into internal display, remember to save it and restart the mac. Use SwitchresX set the internal resolution to the new one. Then you can mirror the exterior display to internal display in display settings. Then it should be hidpi on external display too. I am using m1 macboom air on 11.3 and 11.4 beta on a 34 ultrawide. This should be the solution so far. Hope this can help. Been using like this for weeks.
> > 
> > 
> > What is your external monitor, ie. brand and resolution? Did you have the issue of "Not activate - Invalid"?
> > > Then you can mirror the exterior display to internal display in display settings.
> > 
> > 
> > Can you give more details, screenshots would help. The reason I ask is, I'm using mac mini, there is no internal display. Thanks in advance!
> 
> Mine is LG 34 ultrawide 3440x 1440
> for mac mini it doesn’t have an internal display, so probably it can use a 4k edid emulator to work as an internal display.
> 
> ![93A55AF8-DAE5-4F3D-A059-67BBA25083B7](https://user-images.githubusercontent.com/78505232/116021887-f9220f80-a5fd-11eb-96da-e701d60329de.png)
hey ,
can you work more than 720P ?

**Comment by JM-Mendez** [comment]
@sunjianan9900 I haven't tried on a Mac mini, sorry. Just an macbook air. My display is a Dell ultrawide u3419w

**Comment by JM-Mendez** [comment]
> Just found this solution, would took me out of my panic state of this new M1 purchase :D
> 
> https://github.com/waydabber/BetterDummy/
> 
> Now I can have multi monitors, extended, and with scaled HiDpi - I'm ecstatic.

I can confirm that this not only works, but works absolutely phenomenally. My display is actually scaling properly now 😄 super crip, and no mouse jaggyness.

Thank you @seanbperiod!

**Comment by plamenh** [comment]
Yeah, that's legit. It works great.

**Comment by seanbperiod** [comment]
@remoteweb Ask their support.

**Comment by remoteweb** [comment]
for anyone interested, to extend your desktop you need to create your dummy display (16x9 etc), open Displays, go to Arrangement and press the option key and drag and drop your External screen to the Dummy Display. Then you ll have hidpi external screen that extends your main display.

**Comment by plamenh** [comment]
An alternative guide:

1. Make your laptop display main.
2. Make the Dummy display extended.
3. Mirror the dummy display on your real external display.
